src/vs/workbench/services/textfile/common/textFileEditorModel.ts

```diff
@@ -778,9 +746,9 @@ export class TextFileEditorModel extends BaseTextEditorModel implements ITextFil
 		this.createTextEditorModelPromise = null;
 
 		this.cancelAutoSavePromises();
-		this.cancelBackupPromises();
 
-		this.fileService.discardBackup(this.resource);
+		// TODO: Can this be moved to BackupModelService?
+		this.backupFileService.discardAndDeregisterResource(this.resource);
```

Comment:
Yes, I suggest another event on the service maybe when this model gets disposed.